internal/lockclient/simple_client.go

```diff
@@ -103,3 +133,91 @@ func (sc *SimpleClient) Release(d lockservice.Descriptors) error {
 func (sc *SimpleClient) StartService(cfg Config) error {
 	panic("TODO")
 }
+
+// Checkacquire checks for acquisition of lock and returns the owner if the lock
+// is already acquired.
+// The errors returned can be due to HTTP errors or marshalling errors.
+// A "file is not acquired" error is returned if so and no error and an owner is
+// returned if the object is acquired.
+func (sc *SimpleClient) Checkacquire(d lockservice.ObjectDescriptor) (string, error) {
+	if sc.cache != nil {
+		owner, err := sc.getFromCache(d)
+		if err != nil {
+			return "", err
+		}
+		return owner, nil
+	}
+
+	endPoint := sc.config.IPAddr + ":" + sc.config.PortAddr + "/checkacquire"
+	data := lockservice.LockCheckRequest{FileID: d.ObjectID}
```

Comment:
Is a new type really needed for a `checkAcquire()` request?

Reply:
Yep else we need to send a `""` which just looks bad. Moreover, this is used in multiple places and I think it deserves to be a datatype.